routing/src/adjacency.rs

```diff
@@ -0,0 +1,40 @@
+use crate::interface::IfIndex;
+use net::eth::mac::Mac;
+use std::collections::HashMap;
+use std::net::IpAddr;
+
+#[allow(dead_code)]
+pub struct Adjacency {
+    pub address: IpAddr,
+    pub mac: Mac,
+    pub ifindex: IfIndex,
+}
+pub struct AdjacencyTable(HashMap<IpAddr, Adjacency>);
+
+#[allow(dead_code)]
+impl AdjacencyTable {
+    pub(crate) fn new() -> Self {
+        Self(HashMap::new())
+        // Todo: use a fast hasher
+    }
+}
+
+#[allow(dead_code)]
+impl AdjacencyTable {
+    pub(crate) fn add_adjacency(&mut self, address: IpAddr, mac: Mac, ifindex: IfIndex) {
+        self.0.insert(
```

Comment:
I think  it makes more sense to have this method accept an `Adjacency` directly.

That way `Adjacency` can impose any needed validity constraints in its constructor.

It is also more consistent with the `get_adjacency` method